### ISSUE TITLE
Abstraction for uC/OS OS_EVENT?

### ISSUE THREAD

**Issue by msplr** [cvra/platform-abstraction#10]
Will OS_EVENT's be used?

**Comment by antoinealb** [comment]
Well `OS_EVENT` is a generic type used for mutex, semaphore and various other features.
So what do you mean ?

**Comment by froj** [comment]
> Well `OS_EVENT` is a generic type used for mutex, semaphore and various other features. So what do you mean ?

Not anymore in uC/OS-III.
But still: Could you elaborate?

**Comment by msplr** [comment]
> Well `OS_EVENT` is a generic type used for mutex, semaphore and various other features.

right. I meant `OSEventPendMulti()`. Being able to wait for multiple "events" at once is sometimes very useful, but it is probably difficult to abstract and test this.

> Not anymore in uC/OS-III.

ok, didn't know. I was looking at uC/OS-II documentation.

**Comment by froj** [comment]
> > Not anymore in uC/OS-III.
> 
> ok, didn't know. I was looking at uC/OS-II documentation.

They started to make actual types like `OS_MUTEX` and `OS_SEM`.

`OSEventPendMulti()` seems really useful to me, too.

The new [API](https://doc.micrium.com/display/osiiidoc/OSPendMulti).

**Comment by antoinealb** [comment]
This seems pretty useful and could be abstracted into something like this maybe (function name not appropriate) :

``` c
semaphore_t *os_semaphore_take_many(semaphore_t *semaphores[], size_t count);
```

The function would return the semaphore that was accepted. Now if we want to add message queues into the mix, we probably should define an abstraction for message queues first.
